### PR TITLE
ci: migrate to Orb Tools 12

### PR DESCRIPTION
**Pull request by EricRibeiro**
## Description

This PR updates the Slack Orb to the new major version of Orb Tools. As it contains breaking changes, several modifications were unnecessary to move from v11 to [v12](https://circleci.com/developer/orbs/orb/circleci/orb-tools). 

## Changes

The comprehensive list of changes can be found below.

### Changes in `.circleci/config.yml`:

1. Update the orb-tools version from 11.4 to 12.0.
1. Remove the `orb-tools/publish` job since development versions are no longer necessary.
1. Move the job requirement list from `orb-tools/publish` to `orb-tools/continue`.
1. Rename the `orb-tools/continue` job parameters to comply with the new snake case standard.
   - `orb_name`, `pipeline_number` and `vcs_type`.
1. Add the new `orb_name` parameter in `orb-tools/continue`.

### Changes in `.circleci/test-deploy.yml`:

1. Update the orb-tools version from 11.4 to 12.0.
1. Remove the `slack: circleci/slack@dev:<<pipeline.git.revision>>` line, and replace it with `slack: {}`.
1. Remove the `orb-tools/lint`, `orb-tools/pack`, and `orb-tools/review` jobs. ⚠️ 
1. Rename the `orb-tools/publish` job parameters to comply with the new snake case standard.
   - `orb_name`, `vcs_type`, `pub_type`, `enable_pr_comment` and `github_token`.
1. Change the `orb-tools/pack` filter to trigger only on tagged releases.
 
 ### Changes in `src/examples/full_deployment_sample.yml`:

1. Replace `slack/on-hold` with `slack/on_hold` to comply with the new snake case standard.

 ### Changes in `src/examples/on_hold_notification.yml`:

1. Replace `slack/on-hold` with `slack/on_hold` to comply with the new snake case standard.

---

⚠️: This change is not related to the migration itself. These jobs were redundant in `test-deploy.yml` as they already run in `config.yml`. See [Orb-Template #2](https://github.com/CircleCI-Public/Orb-Template/pull/2/files).

### CODE REVIEW

**Comment by orb-publisher** [comment]
Your development orb(s) have been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/?version=dev:dev:${CIRCLE_SHA1}

**Review by brivu — APPROVED** [review]
Looks good!

File: .circleci/test-deploy.yml [hunk 1]
```diff
@@ -207,7 +207,11 @@ workflows:
             parameters:
               runner: [cimg, mac, alpine, windows]
       - orb-tools/pack:
-          filters: *filters
+          filters:
+            branches:
+              ignore: /.*/
+            tags:
+              only: /^v[0-9]+\.[0-9]+\.[0-9]+$/
```
Comment (KyleTryon) [inline .circleci/test-deploy.yml:214]: Given we use the tag twice, I would anchor/alias it like the other filters

**Comment by KyleTryon** [comment]
> I believe [RC010](https://github.com/CircleCI-Public/orb-tools-orb/blob/master/src/scripts/review.bats#LL192C8-L192C13) _should_ have failed unless ignored. Looking into this.

I was wrong, I assumed we would failed RC010, turns out this orb was already up-to-code on that.

File: .circleci/test-deploy.yml (hunk 1 above)
Comment (KyleTryon) [reply .circleci/test-deploy.yml:214]: Updated the template to do what i mean here: https://github.com/CircleCI-Public/Orb-Template/pull/11/commits/7b37d65d8f9c8064302fc16e373160b3927d2094

**Pull request merged**